Source/WebCore/html/HTMLVideoElement.cpp

```diff
@@ -180,7 +180,7 @@ bool HTMLVideoElement::supportsFullscreen(HTMLMediaElementEnums::VideoFullscreen
     if (!player()->supportsFullscreen())
         return false;
 
-#if PLATFORM(IOS_FAMILY)
+#if PLATFORM(IOS_FAMILY) && !ENABLE(VIDEO_USES_ELEMENT_FULLSCREEN)
```

Comment:
Do we actually need this ifdef at all? Can we just make the code below that checks all the bits work on iOS?

Reply:
Might be possible with clever refactoring, but there's a lot to untangle underneath this code and I'm not seeing an obvious pathway to doing so. We need to avoid running the following on iOS:

```
    if (videoFullscreenMode == HTMLMediaElementEnums::VideoFullscreenModeStandard && !document().settings().fullScreenEnabled())
        return false;
```

`supportsFullScreenForElement` below this code also consults a delegate method in WK1. On WK2, that method consults the bit on settings. Furthermore, both `supportsFullScreenForElement` and `supportsVideoFullscreen` have additional call sites that need to be audited.

We also need to avoid the `!player()->hasVideo()` below `supportsFullScreenForElement` on iOS. 

At best, I think we would just be moving this `#ifdef` into another method.